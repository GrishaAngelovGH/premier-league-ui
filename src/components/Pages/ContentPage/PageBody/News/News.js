const News = () => (
    <div>NEWS</div>
)

export default News