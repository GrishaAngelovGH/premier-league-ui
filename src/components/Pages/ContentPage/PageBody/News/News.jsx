import { news } from 'images'

export const newsItems = [
	{
		title: 'Guardiola rules out new Manchester City signings in January',
		source: 'Goal.com',
		time: '4 days ago',
		image: news['news1']
	},
	{
		title: `'Welbeck said it wasn't a penalty!' â€“ Henderson fumes at VAR after 
        Liverpool are held by Brighton`,
		source: 'Goal.com',
		time: '3 days ago',
		image: news['news2']
	},
	{
		title: 'Chelsea 0-0 Tottenham Hotspur: Spurs return to the top of the Premier League',
		source: 'BBC News',
		time: '2 days ago',
		image: news['news3']
	},
	{
		title: 'Return of supporters to Stamford Bridge | Official Site | Chelsea Football Club',
		source: 'Chelsea FC',
		time: '5 days ago',
		image: news['news4']
	}
]

const News = ({ newsItems }) => (
	<div className='flex flex-col gap-5 items-center mt-3'>
		{
			newsItems.map(v => (
				<div className='flex items-center border-2 border-gray-400 rounded-md text-center p-1 w-1/2' key={v.title}>
					<div className='w-4/6 md:w-5/6'>
						<a href='#/content/news'>{v.title}</a>
						<div className='text-gray-500 text-sm'>
							{v.source} {v.time}
						</div>
					</div>
					<div className='w-2/6 md:w-1/6'>
						<img src={v.image} className='rounded-md' />
					</div>
				</div>
			))
		}
	</div>
)

export default News