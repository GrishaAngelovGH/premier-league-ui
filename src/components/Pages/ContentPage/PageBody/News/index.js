import News, { newsItems } from './News'

export { newsItems }

export default News