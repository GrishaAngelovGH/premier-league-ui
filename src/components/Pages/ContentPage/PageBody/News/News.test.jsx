import { render } from '@testing-library/react'
import AppContext from 'context'

import News from './News'

test('should render News component', () => {
	const newsItems = [
		{
			title: 'Guardiola rules out new Manchester City signings in January',
			source: 'Goal.com',
			time: '4 days ago',
			image: 'news1'
		}
	]

	const view = render(
		<AppContext.Provider value={{ newsItems }}>
			<News />
		</AppContext.Provider>
	)

	expect(view).toMatchSnapshot()
})