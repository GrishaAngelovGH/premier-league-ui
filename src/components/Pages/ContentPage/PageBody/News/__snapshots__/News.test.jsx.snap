// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render News component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-5 items-center mt-3"
      >
        <div
          class="flex items-center border-2 border-gray-400 rounded-md text-center p-1 w-1/2"
        >
          <div
            class="w-4/6 md:w-5/6"
          >
            <a
              href="#/content/news"
            >
              Guardiola rules out new Manchester City signings in January
            </a>
            <div
              class="text-gray-500 text-sm"
            >
              Goal.com
               
              4 days ago
            </div>
          </div>
          <div
            class="w-2/6 md:w-1/6"
          >
            <img
              class="rounded-md"
              src="news1"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-5 items-center mt-3"
    >
      <div
        class="flex items-center border-2 border-gray-400 rounded-md text-center p-1 w-1/2"
      >
        <div
          class="w-4/6 md:w-5/6"
        >
          <a
            href="#/content/news"
          >
            Guardiola rules out new Manchester City signings in January
          </a>
          <div
            class="text-gray-500 text-sm"
          >
            Goal.com
             
            4 days ago
          </div>
        </div>
        <div
          class="w-2/6 md:w-1/6"
        >
          <img
            class="rounded-md"
            src="news1"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
